### PR TITLE
CFY-7680 Validate private and public IPs during install/configure

### PR DESCRIPTION
**Pull request by mcouthon**

### CODE REVIEW

File: cfy_manager/components/validations.py [hunk 1]
```diff
@@ -174,6 +175,31 @@ def _validate_inputs():
             )
 
 
+def _validate_host(host, input_name):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        sock.connect((host, INTERNAL_REST_PORT))
```
Comment (tehasdf) [inline cfy_manager/components/validations.py]: you can also make this shorter and ipv6-compat by using `socket.create_connection`, eg. like this:
```python
try:
  s = socket.create_connection((host, port))
except socket.error:
   # blabla
else:
  s.close()
```

Comment (tehasdf) [inline cfy_manager/components/validations.py]: this must of course be a port that is open, but i can't think of a good candidate right now

File: cfy_manager/components/validations.py [hunk 2]
```diff
@@ -174,6 +175,31 @@ def _validate_inputs():
...
+                port=INTERNAL_REST_PORT
+            )
+        )
+
+
+def _validate_hosts():
+    logger.info('Validating private and public IPs...')
+
+    _validate_host(config[MANAGER][PRIVATE_IP], 'private IP')
+    _validate_host(config[MANAGER][PUBLIC_IP], 'public IP')
```
Comment (tehasdf) [inline cfy_manager/components/validations.py]: this must not raise a fatal error because the public IP isn't necessarily routable from the manager

**Comment by mcouthon** [comment]
Decided that there's no really good way to make sure the IPs are valid. Any heuristic will have false negatives and/or false positives, so it's best to leave it up to the user to make sure the inputs are correct

**Pull request closed without merging**